// @ip-algorithmics/codeforces-io/dist/cjs/generator.js
#!/usr/bin/env node
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var _a, _b, _c, _d, _e;
Object.defineProperty(exports, "__esModule", { value: true });
const fs_1 = __importDefault(require("fs"));
const argv = require('yargs/yargs')(process.argv.slice(2))
    .usage('Usage: $0 <path> [options]')
    .command('path', 'Creates a solution folder using the path')
    .example('$0 ./TotallyAProblemToSolve', 'Creates the folder TotallyAProblemToSolve in the current folder')
    .alias('js', 'useJs')
    .nargs('js', 0)
    .describe('js', 'Use Javascript instead of Typescript')
    .alias('cjs', 'useCjs')
    .nargs('cjs', 0)
    .describe('cjs', 'Use require from CommonJs instead ES6 imports')
    .alias('f', 'file')
    .nargs('f', 1)
    .describe('f', 'File name, defaults to index')
    .alias('c', 'comment')
    .nargs('c', 1)
    .describe('c', 'File name, defaults to index')
    .help('h')
    .alias('h', 'help').argv;
// .epilog('copyright 2021')
const options = {
    path: (_a = argv._[0]) !== null && _a !== void 0 ? _a : './New Solution',
    js: (_b = argv.js) !== null && _b !== void 0 ? _b : false,
    cjs: (_c = argv.cjs) !== null && _c !== void 0 ? _c : false,
    file: (_d = argv.file) !== null && _d !== void 0 ? _d : 'index',
    comment: (_e = argv.comment) !== null && _e !== void 0 ? _e : ''
};
function createFolderRecursively(path) {
    fs_1.default.mkdirSync(path, { recursive: true });
}
function createFile(path, file, content) {
    fs_1.default.writeFileSync(`${path}/${file}`, content);
}
const importLibrary = options.cjs
    ? "const { readline, print, testOutput, console } = require('@ip-algorithmics/codeforces-io');"
    : "import { readline, print, testOutput, console } from '@ip-algorithmics/codeforces-io';";
const file = options.js ? `${options.file}.js` : `${options.file}.ts`;
const content = `// ${options.comment}
${importLibrary}


testOutput();
${options.js ? '' : 'export {};'}
`;
createFolderRecursively(options.path);
createFile(options.path, 'input.txt', '');
createFile(options.path, 'output.txt', '');
createFile(options.path, file, content);
